### ISSUE TITLE
Inconsistency when staling fields

### ISSUE THREAD

**Issue by Jontem** [dividab/gql-cache-patch#3]
When executing a `InvalidateField` patch you risk get a mix of new and old data.

## Scenario:
Feature `A` and `B` has a query which has a response containts the same entity `News;1`.

A query:
```
query {

  news {
    id
    header
  }

}
```

B query:
```
query {

  news {
    id
    header
    serverGeneratedFieldDependentOnHeader
  }

}
```

A mutation is done on the server and returns an `InvalidateField` patch. The field is marked stale in cache. Feature A then refetches with it's query and merges in it's reponse entities in cache which updates id and header field and removes the stale mark on the field.

When feature B then queries the cache the stale marker is gone but the value on server has changed and should be refetched.

## Solution: Invalidate recursive
When an `InvalidateField` patch is applied do it recursively and mark every field in the hierarchy as stale

**Comment by Jontem** [comment]
@jonaskello made some commits(Should have made a branch...) Have a look if you have an opinion.

**Comment by jonaskello** [comment]
For reference, here is a [comparision](https://github.com/dividab/gql-cache-patch/compare/bceeb277938c042d54bc122ae708953d70e19eef...18cde32193b35e6b8faa043f4543f5f315ad803d) of the latest commits.

**Comment by Jontem** [comment]
fixed and 0.5.0 is released
